plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -268,3 +279,159 @@ internal fun VcsInfo.toSpdxDownloadLocation(resolvedRevision: String?): String {
     }
 }
+
+/**
+ * Return the [SpdxFile]s for the package denoted by [id]. The given [nextFileIndex] is incremented by one for each
+ * returned file, and it's respective value is used as a part of the generated SPDX ID of the files.
+ */
+internal fun OrtResult.getSpdxFiles(
+    id: Identifier,
+    licenseInfoResolver: LicenseInfoResolver,
+    sourceCodeOrigin: SourceCodeOrigin,
+    nextFileIndex: AtomicInteger
+): List<SpdxFile> =
+    getFileFindings(id, licenseInfoResolver, sourceCodeOrigin).orEmpty().sortedBy { it.path }.map { fileFindings ->
+        SpdxFile(
+            spdxId = "${SpdxConstants.REF_PREFIX}File-${nextFileIndex.getAndIncrement()}",
+            checksums = listOf(
+                SpdxChecksum(algorithm = SpdxChecksum.Algorithm.SHA1, checksumValue = fileFindings.sha1)
+            ),
+            filename = fileFindings.path,
+            licenseConcluded = SpdxConstants.NOASSERTION,
+            licenseInfoInFiles = fileFindings.licenses.takeIf { it.isNotEmpty() }?.map { it.toString() }
+                ?: listOf(SpdxConstants.NONE),
+            copyrightText = fileFindings.copyrights.sorted().joinToString("\n").takeUnless {
+                it.isBlank()
+            } ?: SpdxConstants.NONE
+        )
+    }
+
+private data class FileFindings(
+    val path: String,
+    val sha1: String,
+    val licenses: Set<SpdxExpression>,
+    val copyrights: Set<String>
+)
+
+/**
+ * Return the [FileFindings] for the package denoted by [id]. The result contains exactly one entry for each file which
+ * has at least one finding and no entry for files without any findings.
+ */
+private fun OrtResult.getFileFindings(
+    id: Identifier,
+    licenseInfoResolver: LicenseInfoResolver,
+    sourceCodeOrigin: SourceCodeOrigin
+): Set<FileFindings>? {
+    val resolvedLicenseInfo = licenseInfoResolver.resolveLicenseInfo(id)
+
+    val licensesByFilePath = resolvedLicenseInfo.getLicensesByFilePath(sourceCodeOrigin) ?: return null
+    val copyrightsByFilePath = resolvedLicenseInfo.getCopyrightsByFilePath(sourceCodeOrigin) ?: return null
+    val sha1ByFilepath = getFileListForId(id)?.takeIf { it.provenance.matches(sourceCodeOrigin) }?.files?.associateBy(
+        { it.path }, { it.sha1 }
+    ) ?: return null
+
+    val findingPaths = copyrightsByFilePath.keys + licensesByFilePath.keys
+
+    return findingPaths.mapNotNullTo(mutableSetOf()) { path ->
+        FileFindings(
+            path = path,
+            // The sha1 is not available in case path is not withing the VCS path.
+            sha1 = sha1ByFilepath[path] ?: return@mapNotNullTo null,
+            licenses = licensesByFilePath[path].orEmpty(),
+            copyrights = copyrightsByFilePath[path].orEmpty()
+        )
+    }
+}
+
+/**
+ * Return the file paths corresponding to all non-excluded, curated copyright findings for the specified
+ * [sourceCodeOrigin] associated with the copyright findings under the respective path, or `null` if the
```

Comment:
Better say "copyright statements" here as the result contains only the statements.